javascript: x86_64-pc-windows-msvc tests are flaky in CI

### What happened?

This test is flaky in CI.

Example: https://github.com/apache/arrow-adbc/actions/runs/23300567140/job/67762101580?pr=4123#logs

It has something to do with how SQLite is being compiled before running the test suite.



### Stack Trace

```
MSBuild version 17.14.40+3e7442088 for .NET Framework

  1>Checking Build System
  Building Custom Rule D:/a/arrow-adbc/arrow-adbc/c/vendor/fmt/CMakeLists.txt
  Building Custom Rule D:/a/arrow-adbc/arrow-adbc/c/vendor/nanoarrow/CMakeLists.txt
  format.cc
  nanoarrow.c
  nanoarrow.vcxproj -> D:\a\arrow-adbc\arrow-adbc\javascript\build\vendor\nanoarrow\Debug\nanoarrow.lib
  Building Custom Rule D:/a/arrow-adbc/arrow-adbc/c/driver/common/CMakeLists.txt
  utils.c
  adbc_driver_common.vcxproj -> D:\a\arrow-adbc\arrow-adbc\javascript\build\driver\common\Debug\adbc_driver_common.lib
  os.cc
  Generating Code...
  fmt.vcxproj -> D:\a\arrow-adbc\arrow-adbc\javascript\build\vendor\fmt\Debug\fmtd.lib
  Building Custom Rule D:/a/arrow-adbc/arrow-adbc/c/driver/framework/CMakeLists.txt
  objects.cc
  utility.cc
  Generating Code...
  adbc_driver_framework.vcxproj -> D:\a\arrow-adbc\arrow-adbc\javascript\build\driver\framework\Debug\adbc_driver_framework.lib
  Building Custom Rule D:/a/arrow-adbc/arrow-adbc/c/driver/sqlite/CMakeLists.txt
  Building Custom Rule D:/a/arrow-adbc/arrow-adbc/c/driver/sqlite/CMakeLists.txt
  sqlite.cc
  sqlite.cc
  statement_reader.c
LINK : fatal error LNK1201: error writing to program database 'D:\a\arrow-adbc\arrow-adbc\javascript\build\driver\sqlite\Debug\adbc_driver_sqlite.pdb'; check for insufficient disk space, invalid path, or insufficient privilege [D:\a\arrow-adbc\arrow-adbc\javascript\build\driver\sqlite\adbc_driver_sqlite_shared.vcxproj]
  statement_reader.c
  adbc_driver_sqlite_static.vcxproj -> D:\a\arrow-adbc\arrow-adbc\javascript\build\driver\sqlite\Debug\adbc_driver_sqlite.lib
```

### How can we reproduce the bug?

_No response_

### Environment/Setup

_No response_

--- a/ci/scripts/node_build.sh
+++ b/ci/scripts/node_build.sh
@@ -37,9 +37,10 @@ cmake -S "${REPO_ROOT}/c" -B "${OUT_DIR}" \
   -DCMAKE_BUILD_TYPE=RelWithDebInfo \
   -DADBC_DRIVER_SQLITE=ON \
   -DADBC_BUILD_TESTS=OFF \
-  -DADBC_BUILD_SHARED=ON
+  -DADBC_BUILD_SHARED=ON \
+  -DADBC_BUILD_STATIC=OFF
 
-cmake --build "${OUT_DIR}" --target install -j
+cmake --build "${OUT_DIR}" --config RelWithDebInfo --target install -j
 
 # 2. Build Node.js Package
 echo "=== Building Node.js Package ==="

